wrapanapi/systems/redfish.py

```diff
@@ -35,3 +53,80 @@ def info(self):
 
     def __del__(self):
         """Disconnect from the API when the object is deleted"""
+
+    def server_stats(self, *requested_stats, **kwargs):
+        # Get the requester which represents the class of this method's caller
+        requester = kwargs.get('requester')
+
+        # Retrieve and return the stats
+        requested_stats = requested_stats or self._stats_available
+
+        return {stat: self._server_stats_available[stat](self, requester)
+                for stat in requested_stats}
+
+    def server_inventory(self, *requested_items, **kwargs):
+        # Get the requester which represents the class of this method's caller
+        requester = kwargs.get('requester')
+
+        # Retrieve and return the inventory
+        requested_items = requested_items or self._server_inventory_available
+
+        return {item: self._server_inventory_available[item](self, requester)
+                for item in requested_items}
+
+    def get_server(self, server_name):
+        """
+        Fetch a Redfish resource that represents a particular physical server.
+
+        Note: ideally, we would have used a Redfish oid of the server that the
+        Redfidh provider is currently storing and showing in the GUI labeled
+        with "UUID". However, we are not clear with how to get that information
+        from the appliance.
+        """
+        for node in self.api_client.Systems.Members:
+            # Redfish representation of a computation system doesn't have a
+            # name, and the provider uses the Id of the system as a name
+            if node.Id == str(server_name):
+                return node
```

Comment:
Redfish API client already handles caching for us, so calling `self.api_client.Systems.Members` will only fetch data from remote on first use.

Reply:
👍 fancy :)

@matejart in such case I suggest we override `disconnect` method on RedfishSystem and clear cache in it. Not necessarily in this PR though.

Reply:
When the garbage collector destroys the RedfishSystem instance, it will also destroy the client and the along with it the cache. Hence, the `__del__` method is redundant. I don't think any explicit disconnecting is needed.

Reply:
Ye ye, I was thinking more like if you use it twice:

```python
mgmt = provider.get_mgmt()
mgmt.validate_state()

# create new server or something

mgmt.validate_state() #<--- this one will use cached list of servers so it will fail
```

Unless @tadeboro if the client is smart enough to automatically update the cache when new server is created?